crates/ra_hir_ty/src/diagnostics.rs

```diff
@@ -159,3 +162,31 @@ impl AstDiagnostic for BreakOutsideOfLoop {
         ast::Expr::cast(node).unwrap()
     }
 }
+
+#[derive(Debug)]
+pub struct MissingUnsafe {
+    pub file: HirFileId,
+    pub fn_def: AstPtr<ast::FnDef>,
```

Comment:
I think this should point at the particular unsafe operation, not at the function. (and the message should go more in the direction of "This operation is unsafe and requires unsafe function or block", like in rustc)

Reply:
Done